File: app/src/main/java/com/gproust/sprout/ui/baby/BabyScreen.kt
package com.gproust.sprout.ui.baby

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableLongStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.gproust.sprout.R
import com.gproust.sprout.data.local.BreastSide
import com.gproust.sprout.ui.common.EmptyHint
import com.gproust.sprout.ui.common.SproutTopBar
import com.gproust.sprout.ui.common.babyAge
import com.gproust.sprout.ui.home.BabyPane
import com.gproust.sprout.ui.home.GrowthSpurtNote
import com.gproust.sprout.ui.home.HomeViewModel
import com.gproust.sprout.ui.rememberSproutViewModelFactory
import kotlinx.coroutines.delay

/** How often the "how long ago" chips are refreshed. */
private const val CHIP_REFRESH_MS = 60_000L

/**
 * One baby on their own: the same pane the dashboard shows in place when there
 * is only one child, given a tab of its own once there is a choice to make.
 *
 * Which baby this is stays the active-baby selection (BDR-3) — opening a card
 * on the dashboard sets it before navigating here, so the two can't disagree.
 */
@Composable
fun BabyScreen(
    onNavigate: (String) -> Unit,
    onQuickFeed: (BreastSide) -> Unit,
) {
    val vm: HomeViewModel = viewModel(factory = rememberSproutViewModelFactory())
    val state by vm.uiState.collectAsState()
    val activeId by vm.activeBabyId.collectAsState()
    val context = LocalContext.current

    var now by remember { mutableLongStateOf(System.currentTimeMillis()) }
    LaunchedEffect(Unit) {
        while (true) {
            delay(CHIP_REFRESH_MS)
            now = System.currentTimeMillis()
        }
    }

    // Falls back to the first tracked baby: a selection can go stale when the
    // baby it named is archived or deleted from another screen.
    val summary = state.babies.firstOrNull { it.baby.id == activeId }
        ?: state.babies.firstOrNull()

    Scaffold(
        topBar = {
            SproutTopBar(summary?.baby?.name ?: stringResource(R.string.nav_baby))
        },
    ) { padding ->
        if (summary == null) {
            EmptyHint(
                stringResource(R.string.home_setup_prompt),
                Modifier.padding(padding).padding(16.dp),
            )
            return@Scaffold
        }

        Column(
            Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
        ) {
            BabyPane(
                summary = summary,
                tracksWellbeing = state.tracksWellbeing,
                now = now,
                onFeed = onQuickFeed,
                onNavigate = onNavigate,
                header = {
                    Text(
                        babyAge(context, summary.baby.birthDate, now),
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.padding(bottom = 12.dp),
                    )
                },
            )
            state.spurts[summary.baby.id]?.let {
                Spacer(Modifier.height(16.dp))
                GrowthSpurtNote(it)
            }
        }
    }
}

File: ios/Sprout/Sources/Screens/Baby/BabyScreen.swift
import SproutData
import SwiftUI

/// One baby on their own: the same pane the dashboard shows in place when there
/// is only one child, given a tab of its own once there is a choice to make.
///
/// Which baby this is stays the active-baby selection (BDR-0003) — opening a
/// card on the dashboard sets it before navigating here, so the two cannot
/// disagree.
struct BabyScreen: View {
    let model: HomeViewModel
    let onOpen: (LogDestination) -> Void

    var body: some View {
        Group {
            if let summary = model.selectedBaby {
                ScrollView {
                    BabyPane(
                        summary: summary,
                        tracksWellbeing: model.tracksWellbeing,
                        now: model.now,
                        onFeed: { model.startFeed(for: summary.baby, on: $0); onOpen(.feeding) },
                        onOpen: onOpen,
                        onShareRecord: { onOpen(.report) },
                        header: {
                            Text(SproutFormat.age(birthDate: summary.baby.birthDate, now: model.now).text)
                                .font(.callout)
                                .foregroundStyle(SproutColor.onSurfaceVariant)
                                .frame(maxWidth: .infinity, alignment: .leading)
                        }
                    )
                    .padding(Spacing.regular)
                }
            } else {
                EmptyHint(Str.t("home_setup_prompt"))
            }
        }
        .navigationTitle(model.selectedBaby?.baby.name ?? Str.t("nav_baby"))
        .navigationBarTitleDisplayMode(.inline)
        .sproutStyle()
    }
}

/// The parent's own tab.
///
/// Pumping and wellbeing are the two logs in Sprout that belong to a *person*
/// rather than to a child — neither carries a `babyId`, by decision, and neither
/// is deleted when a baby is (BDR-0001, BDR-0007). They used to sit in the
/// dashboard's log list next to the baby's screens with nothing to say why, and
/// the daily check-in had nowhere at all once it was dismissed. This is that
/// shelf.
struct YouScreen: View {
    let model: HomeViewModel
    let onOpen: (LogDestination) -> Void

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: Spacing.snug) {
                Text(Str.t("you_intro"))
                    .font(.callout)
                    .foregroundStyle(SproutColor.onSurfaceVariant)
                    .padding(.bottom, Spacing.tight)

                YouEntry(
                    systemImage: "drop.triangle.fill",
                    title: Str.t("screen_pumping"),
                    detail: Str.t("you_pumping_body")
                ) { onOpen(.pumping) }

                // Hidden for a parent who has turned their own tracking off, the
                // same way the log grid drops it — the history stays, untouched.
                if model.tracksWellbeing {
                    YouEntry(
                        systemImage: "heart.fill",
                        title: Str.t("screen_wellbeing"),
                        detail: Str.t("you_wellbeing_body")
                    ) { onOpen(.wellbeing) }

                    YouEntry(
                        systemImage: "heart.text.square.fill",
                        title: Str.t("home_checkin_title"),
                        detail: model.checkInPending
                            ? Str.t("home_checkin_body")
                            : Str.t("you_checkin_done")
                    ) { onOpen(.checkIn) }
                }
            }
            .padding(Spacing.regular)
        }
        .navigationTitle(Str.t("nav_you"))
        .navigationBarTitleDisplayMode(.inline)
        .sproutStyle()
    }
}

private struct YouEntry: View {
    let systemImage: String
    let title: String
    /// Not `body` — that name is already taken by the view's own, and a stored
    /// property called `body` shadows it into a compile error.
    let detail: String
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: Spacing.snug + 2) {
                Image(systemName: systemImage)
                    .font(.title3)
                    .foregroundStyle(SproutColor.primary)
                    .frame(width: 24)
                VStack(alignment: .leading, spacing: 2) {
                    Text(title)
                        .font(.subheadline.weight(.semibold))
                        .foregroundStyle(SproutColor.onSurface)
                    Text(detail)
                        .font(.callout)
                        .foregroundStyle(SproutColor.onSurfaceVariant)
                        .multilineTextAlignment(.leading)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                Image(systemName: "chevron.right")
                    .foregroundStyle(SproutColor.onSurfaceVariant)
            }
            .padding(Spacing.regular)
            .background(
                RoundedRectangle(cornerRadius: Radius.card)
                    .stroke(SproutColor.outline.opacity(0.4), lineWidth: 1)
            )
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }
}
